unit-tests/src/test/java/org/eclipse/collections/impl/list/immutable/AbstractImmutableListTestCase.java

```diff
@@ -429,28 +430,40 @@ public void toSortedListBy()
         Assert.assertEquals(this.classUnderTest(), sortedList);
     }
 
-    @Test
+    @Test(expected = UnsupportedOperationException.class)
     public void removeAtIndex()
     {
-        Verify.assertThrows(UnsupportedOperationException.class, () -> this.classUnderTest().castToList().remove(1));
+        this.classUnderTest().castToList().remove(1);
     }
 
-    @Test
+    @Test(expected = UnsupportedOperationException.class)
     public void set()
     {
-        Verify.assertThrows(UnsupportedOperationException.class, () -> this.classUnderTest().castToList().set(0, 1));
+        this.classUnderTest().castToList().set(0, 1);
     }
 
-    @Test
+    @Test(expected = UnsupportedOperationException.class)
```

Comment:
I would prefer to keep using Verify.assertThrows() in favor of @Test(expected = class). It works better with multiple assertions in a method, and it works better when delegating to super.sameTestMethod().

Reply:
Still relevant I think.

Reply:
Fixed.